libmscore/rendermidi.cpp

```diff
@@ -439,49 +532,449 @@ static void collectMeasureEvents(EventMap* events, Measure* m, Staff* staff, int
+            tickInc = 2;
+      else
+            tickInc = 1;
+
+      // See these functions graphically at: https://www.desmos.com/calculator/kk89ficmjk
+      std::function<int(int,int,int)> valueFunction;
+      switch (changeMethod) {
+            case VeloChangeMethod::EXPONENTIAL:
+                  // Due to the nth-root, exponential functions do not flip with negative values, and cause errors,
+                  // so treat it as a piecewise function.
+                  if (exprDiff > 0)
+                        valueFunction = [](int d, int ct, int tt) { return int(pow(pow((d+1), 1.0/double(tt)), double(ct))-1); };
+                  else
+                        valueFunction = [](int d, int ct, int tt) { return -int(pow(pow((-d+1), 1.0/double(tt)), double(ct))+1); };
+                  break;
+            // Uses sin x transformed, which _does_ flip with negative numbers
+            case VeloChangeMethod::EASE_IN_OUT:
+                  valueFunction = [](int d, int ct, int tt) { return int((double(d)/2.0)*(sin(double(ct)*(M_PI/double(tt))-M_PI/2.0)+1)); };
+                  break;
+            case VeloChangeMethod::EASE_IN:
+                  valueFunction = [](int d, int ct, int tt) { return int(double(d)*(sin(double(ct-tt)*(M_PI/double(2*tt)))+1)); };
+                  break;
+            case VeloChangeMethod::EASE_OUT:
+                  valueFunction = [](int d, int ct, int tt) { return int(double(d)*sin(double(ct)*(M_PI/double(2*tt)))); };
+                  break;
+            case VeloChangeMethod::NORMAL:
+            default:
+                  // We can calculate how to increase the ticks, since it is linear
+                  tickInc = exprTicks/abs(exprDiff);
+                  valueFunction = [](int d, int ct, int tt) { return int(d*(double(ct)/double(tt))); };
+                  break;
+            }
+
+      // prevent possible infinite loop
+      if (tickInc < 1)
+            tickInc = 1;
+
+      for (int i=stick; i < etick; i+=tickInc) {
+            int valueToAdd = valueFunction(exprDiff, i-stick, exprTicks);
+            int exprVal = startExpr + valueToAdd;
+            NPlayEvent cc11event = NPlayEvent(ME_CONTROLLER, channel, controller, abs(exprVal));
+            events->insert(std::pair<int, NPlayEvent>(i, cc11event));
+            //qDebug() << "added cc event val: " << exprVal << " at tick " << i;
+            }
+      }
+
+//---------------------------------------------------------
+//   collectMeasureEventsDefault
+//    this uses only CC events to control note velocity, and sets the
+//    note-on velocity to always be 127 (max). This is the best method
+//    of single note dynamics, but only works if the soundfont supports it.
+//    Method is one of:
+//          -1 - default: velocity is fixed at 127
+//          -2 - note-on velocity is the same as the start velocity of the seg
+//          [value]     - note on velocity is fixed at this value (not used by default)
```

Comment:
I would suggest one of the two options here:
1) Rename the variable to `velocity` and make 127 its default value. That way we need only one special value for it instead of two.
2) Probably rename the variable either but have also symbolic constants for those special values. It would be not the best option for a code readability to have some function calls with seemingly mysterious `-2` as an argument.

Reply:
Fixed.